extensions/parquet/base/src/main/java/io/deephaven/parquet/base/ParquetFileReader.java

```diff
@@ -29,17 +27,16 @@ public class ParquetFileReader {
     static final byte[] MAGIC = MAGIC_STR.getBytes(StandardCharsets.US_ASCII);
     public final FileMetaData fileMetaData;
     private final SeekableChannelsProvider channelsProvider;
-    private final ThreadLocal<CodecFactory> codecFactory;
+    private static final DeephavenCodecFactory codecFactory = new DeephavenCodecFactory(Collections.emptyList());
```

Comment:
This should probably be a singleton, shared at least with ParquetFileWriter (and possible as an instance in DeephavenCodecFactory itself), with the list of classes to also reference loaded from Configuration?

Putting it in DeephavenCodecFactory would require making that module depend on :Configuration, not the end of the world

Reply:
Yep, just didnt want to tackle it until any other opinions had been expressed.